File: android/app/src/main/java/voidawt/DisplayMode.java
package voidawt;

public class DisplayMode {
    public static final int BIT_DEPTH_MULTI = -1;
    public static final int REFRESH_RATE_UNKNOWN = 0;

    private final int width;
    private final int height;
    private final int bitDepth;
    private final int refreshRate;

    public DisplayMode(int width, int height, int bitDepth, int refreshRate) {
        this.width = width;
        this.height = height;
        this.bitDepth = bitDepth;
        this.refreshRate = refreshRate;
    }

    public int getWidth() {
        return width;
    }

    public int getHeight() {
        return height;
    }

    public int getBitDepth() {
        return bitDepth;
    }

    public int getRefreshRate() {
        return refreshRate;
    }

    public boolean equals(Object obj) {
        if (!(obj instanceof DisplayMode)) {
            return false;
        }
        DisplayMode other = (DisplayMode) obj;
        return width == other.width && height == other.height && bitDepth == other.bitDepth && refreshRate == other.refreshRate;
    }
}

File: src/jsMain/kotlin/awt/DisplayMode.kt
package awt

actual class DisplayMode actual constructor(
    private val width: Int,
    private val height: Int,
    private val bitDepth: Int,
    private val refreshRate: Int
) {
    actual fun getWidth(): Int = width
    actual fun getHeight(): Int = height
    actual fun getBitDepth(): Int = bitDepth
    actual fun getRefreshRate(): Int = refreshRate

    override fun equals(other: Any?): Boolean =
        other is DisplayMode && width == other.width && height == other.height &&
            bitDepth == other.bitDepth && refreshRate == other.refreshRate

    override fun hashCode(): Int =
        ((width * 31 + height) * 31 + bitDepth) * 31 + refreshRate
}
